### DISCUSSION TITLE
安卓播放pag视频，同时播放音乐文件，音乐和视频不同步

### DISCUSSION THREAD

**Discussion by ereters** [Tencent/libpag#1759] [General] [answered]
安卓播放pag视频，同时播放过音乐文件，音乐和视频不同步，pag视频要慢些，请问是什么问题？ 大概两句代码：
 mViewBinding.lavPagControlView.play()//播放pag视频
player.gsyVideoManager.start()//播放音乐文件

**Comment by domchen** [comment, marked as answer]
音画同步是要自己做的。系统的音频播放本来就会有延迟，你找下都有专门的接口查询会延迟多少毫秒。要是你的音频不是简单的背景音乐，有严格的对齐关系，要自己做音画对齐。一般以音频为准来通过setProgress更新PAGView的进度。这个和PAG渲染没什么关系，是业务要做的逻辑。音视频播放不属于标准能力，还挺复杂的。我们在PAG企业版里也提供了开箱即用的音频播放附加能力，会自动对齐时间轴，还可以替换视频作为占位图，以及导出视频。具体可以参考：https://pag.art/feature.html#comparison  如果只想用免费的社区版，这部分要自己开发。
